sdk/cognitiveservices/cognitiveservices-qnamaker-runtime/swagger/readme.md

```diff
@@ -0,0 +1,25 @@
+# Cognitive Services QnAMaker Runtime SDK
```

Comment:
Do we usually commit this file to the repo?
Am afraid it will get deleted and create noise in future PRs

Reply:
I added this to make future re-generations easier and consistent. This is mainly to be able to run `autorest --typescript` without any other parameters

Given the current re-generation config, this file won't be deleted.

I thought this would make the process easier especially since the specs repo have the 2 swaggers under the same "folder" and we're generating a package for each. However if you think it'd be better to document these configurations in a different place I'll be happy to move it

Reply:
Looping in @jeremymeng, @xirzec and @HarshaNalluru who have done re-generating of the libraries quite a few times

Reply:
Ideally the transform would live together with the swagger file. I believe that was what we planned to do with the storage swagger readme.md too.  But it might take a while to get there.  Is it not feasible to update https://github.com/Azure/azure-rest-api-specs/blob/796f8ca8f9760cbd32cd3117496d6d53d9858608/specification/cognitiveservices/data-plane/QnAMaker/readme.typescript.md?

Reply:
I haven't seen this pattern for track1, but for track2 I really do prefer it since as @joheredi mentions, it's a pain to get the exact arguments right.

Reply:
Thanks Jeremy! the problem I'm facing with the readme in `azure-rest-api-specs`, is that , they are batching the generation of [2 swaggers](https://github.com/Azure/azure-rest-api-specs/tree/master/specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0) (authoring and runtime).
![image](https://user-images.githubusercontent.com/21109913/73307327-d7579400-41d2-11ea-8bc5-a5aa86ecae50.png) ^From [README.md](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/cognitiveservices/data-plane/QnAMaker/readme.md)

C# supports having 2 clients within a single namespace, and this batching accomplishes that. However, Autorest.Typescript doesn't support having 2 clients in the same package.



When running autorest.typescript against the readme in the `azure-rest-api-specs` repository, the files generated by the first pass (qnaMaker) get overwritten by the second pass (qnaMalerRuntime).

To overcome this limitation, I'm generating a separate package for each client. So I thought about adding this config here to 1) not break other languages that support multiple clients per package and 2) to have an easier way to remember the config to generate these.

Other options I thought of:
1) Split the take out `QnAMakerRuntime.json` from `azure-rest-api-specs/blob/master/specification/cognitiveservices/data-plane/QnAMaker` into a new `..../data-plane/QnAMakerRuntime` folder with its own README.md
   - The problem with this option is that it would break the generation for C# (and other languages that expect a batched run) since they expect both to be in the same place and to come from a batched autorest run.
   - This won't be a problem if all other languages generate a separate package per Swagger

2) Just use command-line flags for generation
   - The problem with this is that it will be hard to remember the flags needed for these libraries in the future. If we document these in another place it may create extra overhead of finding it for the person that is executing the task


I'm open to suggestions :smile:

Reply:
I am fine with current solution. It's unfortunate that we could not maintain the transform and the swagger in a same place, and share among languages.

Reply:
Okay so I'll leave these config files for the sake of easier and consistent future re-gens, thanks for the input guys :smile: